File: app/src/main/java/com/devdream/cookall/core/context/AppContext.java
package com.devdream.cookall.core.context;

import android.app.Application;
import android.content.Context;

public class AppContext extends Application {

    private static Context context;

    public void onCreate() {
        super.onCreate();
        AppContext.context = getApplicationContext();
    }

    public static Context getContext() {
        return AppContext.context;
    }

}

File: app/src/main/java/com/devdream/cookall/context/AppContext.kt
package com.devdream.cookall.context

import android.app.Application
import android.content.Context

class AppContext : Application() {

    init {
        instance = this
    }

    companion object {

        private var instance: AppContext? = null

        fun applicationContext() : Context {
            return instance!!.applicationContext
        }
    }

    override fun onCreate() {
        super.onCreate()

        val context: Context = AppContext.applicationContext()
    }

}
